craft_application/commands/lifecycle.py

```diff
@@ -420,6 +422,55 @@ def _run(
         return self._run_real(parsed_args=parsed_args, step_name=step_name)
 
 
+class TestCommand(PackCommand):
+    """Command to run project tests.
+
+    The test command invokes the spread command with a processed spread.yaml
+    configuration file.
+
+    This command is opt-in for applications in craft-application 5 and will become
+    a standard lifecycle command in a future major release.
+    """
+
+    name = "test"
+    help_msg = "Run project tests"
+    overview = textwrap.dedent(
+        """
+        Run spread tests for the project.
+        """
+    )
+    common = True
+
+    @override
+    def _fill_parser(self, parser: argparse.ArgumentParser) -> None:
+        # Skip the parser additions that `pack` adds.
```

Comment:
at some point we might want to revisit this, `spread -debug` and `spread -shell-after` are quite useful
but I agree with leaving it out for now

Reply:
Agreed. https://github.com/canonical/craft-application/issues/696